staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -182,25 +204,44 @@ func (tr *TypedConfigObjectReference) String() string {
 }
 
 // rmSetup is invoked at startup to create the infrastructure of this filter
-func rmSetup(loopbackClientConfig *restclient.Config, serverConcurrencyLimit int, requestWaitLimit time.Duration) *requestManagement {
+func rmSetup(clientConfig *restclient.Config, serverConcurrencyLimit int, requestWaitLimit time.Duration) *requestManagement {
+	kubeClient, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		klog.Errorf("Failed to construct Kubernetes client: %s", err.Error())
+		return nil
+	}
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, 0)
+	fci := kubeInformerFactory.Flowcontrol().V1alpha1()
+	pli := fci.PriorityLevelConfigurations()
+	fsi := fci.FlowSchemas()
 	reqMgmt := &requestManagement{
 		configQueue:            workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "req_mgmt_config_queue"),
+		plInformer:             pli.Informer(),
+		plLister:               pli.Lister(),
+		fsInformer:             fsi.Informer(),
+		fsLister:               fsi.Lister(),
 		serverConcurrencyLimit: serverConcurrencyLimit,
 		requestWaitLimit:       requestWaitLimit,
 	}
+	if !reqMgmt.initialSync() {
+		return nil
+	}
 	// TODO: finish implementation
 	return reqMgmt
 }
 
 // WithRequestManagement limits the number of in-flight requests in a fine-grained way
 func WithRequestManagement(
 	handler http.Handler,
-	loopbackClientConfig *restclient.Config,
+	clientConfig *restclient.Config, // or should it be the kubeClient??
```

Comment:
I think this should be a kubeClient so that we can test more easily.  Having the ability to pass in fake clients using the `k8s.io/client-go/kubernetes/fake` package (particularly `fake.NewSimpleClientset()`) should make testing easier.

Reply:
This is going to be used in our generalization of https://github.com/kubernetes/kubernetes/blob/c7f9dd0bafe2f3328148a85eed9c18322b9f308e/staging/src/k8s.io/apiserver/pkg/server/config.go#L584 .  I do not see a kubeClient handy there.